docs/OIDC-Provider-Examples/sap-public-oidc-provider.md

```diff
@@ -4,33 +4,54 @@ SAP IAS can be setup as an [OIDC provider](https://help.sap.com/viewer/6d6d63354
+1. Create an OIDC application and set it with [type public](https://help.sap.com/viewer/6d6d63354d1242d185ab4830fc04feb1/Cloud/en-US/a721157cd40544eb9bad40085cf8ec15.html)
+   * in Trust / OpenID Configuration / Grant Types / Authorization Code Flow / Enforce PKCE (S256)
+3. Register the "Redirect URIs" in the application section "OpenID Connect Configuration"
+   * Add following URI in list field:
+   `https://{UAA_HOST}/login/callback/{origin}`. [Additional documentation for achieving this can be found here](https://help.sap.com/viewer/6d6d63354d1242d185ab4830fc04feb1/Cloud/en-US/1ae324ee3b2d4a728650eb022d5fd910.html).
+   * E.g. for a UAA part of a CF-Deployment, this is `https://login.cf.<domain>/login/callback/{origin}`
+   * `{origin}` - is the id you of the OIDC provider you will use in UAA in the next step
    
-2. Copy client id.
+2. Go to "Client Authentication" section and check "Allow Public Client Flows".
+   * This will generate the "client id" on the top of the page
+   * Copy "client id", to use for the uaa configuration.
 
-3. Minimal OIDC configuration needs to be added in login.ym.
-   Read configuration refer to '[https://<tenant ID>.accounts.ondemand.com/.well-known/openid-configuration](https://help.sap.com/viewer/6d6d63354d1242d185ab4830fc04feb1/Cloud/en-US/c297516bae4547eb82eeed80fea2b937.html)' for discoveryUrl and issuer
+4. Minimal OIDC configuration needs to be added in `uaa.yml` or `login.yml` (depending on the setup).
+   Read configuration refer to '[https://<tenant ID>.accounts.ondemand.com/.well-known/openid-configuration](https://help.sap.com/viewer/6d6d63354d1242d185ab4830fc04feb1/Cloud/en-US/c297516bae4547eb82eeed80fea2b937.html)' for discoveryUrl and issuer. E.g. in the example below `ias.public` was selected as `{origin}`
 
         login:
           oauth:
             providers:
               ias.public:
                 type: oidc1.0
-                discoveryUrl: https://trailaccount.accounts.ondemand.com/.well-known/openid-configuration
-                issuer: https://trailaccount.accounts.ondemand.com
+                discoveryUrl: https://<ias_tenant_id>.accounts.ondemand.com/.well-known/openid-configuration
+                issuer: https://<ias_tenant_id>.accounts.ondemand.com
                 scopes:
                   - openid
                   - email
                   - profile
                 linkText: Login with IAS-Public
                 showLinkText: true
-                relyingPartyId: 3feb7ecb-d106-4432-b335-aca2689ad123
+                relyingPartyId: <client_id>
+                addShadowUserOnLogin: true
+
+6. Ensure that the scope `openid`, `email` and `profile` is included in the `scopes` property. Then UAA shadow user (if addShadowUserOnLogin=true) is created with all properties. 
 
-4. Ensure that the scope `openid`, `email` and `profile` is included in the`scopes` property. Then UAA shadow user (if addShadowUserOnLogin=true) is created
```

Comment:
addShadowUserOnLogin: true 
is only needed if you have no user in UAA

So you can also use SAP IPS to provision the user into UAA

remark: the addShadowUserOnLogin is a security setting. With true you allow ALL users from a provider and no pre-provision needed (easy part) , but with false you allow only the pre-provisioned users (which is the more secure way)

Reply:
i think for an initial configuration it is better to have the users to be created by default, as  having to do this manually or via SAP IPS is more complicated